packages/angular_devkit/build_angular/src/builders/jest/index.ts

```diff
@@ -0,0 +1,164 @@
+      commonChunk: false,
+      optimization: false,
+      buildOptimizer: false,
+      sourceMap: {
+        scripts: true,
+        styles: false,
+        vendor: true,
+      },
+    });
+    if (!buildResult.success) {
+      return buildResult;
+    }
+
+    // Execute Jest on the built output directory.
+    const jestProc = execFile('node', [
+      '--experimental-vm-modules',
+      jest,
+
+      `--rootDir="${testOut}"`,
+      '--testEnvironment=jsdom',
+
+      // TODO(dgp1130): Enable cache once we have a mechanism for properly clearing / disabling it.
+      '--no-cache',
+
+      // Run basically all files in the output directory, any excluded files were already dropped by the build.
+      `--testMatch="<rootDir>/**/*.mjs"`,
+
+      // Load polyfills and initialize the environment before executing each test file.
+      // IMPORTANT: Order matters here.
+      // First, we execute `jest-global.mjs` to initialize the `jest` global variable.
+      // Second, we execute user polyfills, including `zone.js` and `zone.js/testing`. This is dependent on the Jest global so it can patch
+      // the environment for fake async to work correctly.
+      // Third, we initialize `TestBed`. This is dependent on fake async being set up correctly beforehand.
+      `--setupFilesAfterEnv="<rootDir>/jest-global.mjs"`,
+      `--setupFilesAfterEnv="<rootDir>/polyfills.mjs"`,
+      `--setupFilesAfterEnv="<rootDir>/init-test-bed.mjs"`,
+
+      // Don't run any infrastructure files as tests, they are manually loaded where needed.
+      `--testPathIgnorePatterns="<rootDir>/jest-global\\.mjs"`,
+      `--testPathIgnorePatterns="<rootDir>/polyfills\\.mjs"`,
+      `--testPathIgnorePatterns="<rootDir>/init-test-bed\\.mjs"`,
+
+      // Skip shared chunks, as they are not entry points to tests.
+      `--testPathIgnorePatterns="<rootDir>/chunk-.*\\.mjs"`,
+
+      // Optionally enable color.
+      ...(colors.enabled ? ['--colors'] : []),
+    ]);
+
+    // Stream test output to the terminal.
+    jestProc.child.stdout?.on('data', (chunk) => {
+      context.logger.info(chunk);
+    });
+    jestProc.child.stderr?.on('data', (chunk) => {
+      context.logger.info(chunk);
+    });
+
+    try {
+      await jestProc;
+    } catch (error) {
+      // No need to propagate error message, already piped to terminal output.
+      return { success: false };
+    }
+
+    return { success: true };
+  },
+);
+
+async function build(
+  context: BuilderContext,
+  options: BrowserEsbuildOptions,
+): Promise<BuilderOutput> {
+  try {
+    for await (const _ of buildEsbuildBrowserInternal(options, context)) {
+      // Nothing to do for each event, just wait for the whole build.
+    }
+
+    return { success: true };
+  } catch (err) {
+    return {
+      success: false,
+      error: (err as Error).message,
+    };
+  }
+}
+
+/** Safely resolves the given Node module string. */
+function resolve(module: string): string | undefined {
```

Comment:
Nit: I suggest to rename it to something more specific like `resolveModule` to avoid conflict with path `resolve`.